### PR TITLE
Psi experiments

### PR DESCRIPTION
**Pull request by eriktaubeneck**
opening a PR to be able to make comments on the code directly

### CODE REVIEW

File: Compiler/bitonic.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    def _(x):
+        k = 2**x
+        @for_range(x)
+        def _(r): 
+            j = 2**r
+            @for_range(n)
+            def _(i):
+                l = i^j #bitwise XOR
+                if (l>i):
+                    if ((i&k) ==0) and (arr[i] > arr[l]) or ((i&k)!=0 and (arr[i] < arr[l])):
```
Comment (eriktaubeneck) [inline Compiler/bitonic.py:52]: is arr a mp-spdz provided `Array` which contains `sints`? if so, can we use comparators and if statements like this, or do we need to use the `sint.if_else` function?

File: Compiler/bitonic.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+
+def bitoniciter(arr, n_threads=None): 
+    '''
+    Parameters
+    ----------
+    arr : an array
+    
+    Does a non-recursive bitonic sort
+    Returns
```
Comment (eriktaubeneck) [inline Compiler/bitonic.py:35]: it doesn't look like this function actually returns anything.

Comment (haqueabida) [reply Compiler/bitonic.py:35]: You're right; it does an in-place swap within arr so there's no need to return anything. I can edit the comment.

Comment (haqueabida) [reply Compiler/bitonic.py:35]: Actually, this function isn't even used at all -- I use bitoniccomps and bitonicmerge_uneven (which are also found inside library). I just used bitonic.py as a practice file, so the final versions (inside library.py) can be used instead. I can actually delete the entire file.

File: Compiler/bitonic.py (hunk 1 above)
Comment (haqueabida) [reply Compiler/bitonic.py:52]: I am going to remove this file; this function is not used. We'll look at library.py from now on.

**Pull request merged**